### PR TITLE
Add cli script

### PR DESCRIPTION
**Pull request by alexandru-elisei**
Specifying a target directory is not yet implemented, pending the result of #22

### CODE REVIEW

File: cli/generate.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+list($options, $unrecognized) = cli_get_params(array(
+    'recipe' => '',
+    'loglevel' => 'WARNING',
+    'help' => '',
+),
+array(
+    'r' => 'recipe',
+    'v' => 'loglevel',
+    'h' => 'help'
+));
```
Comment (mudrd8mz) [inline cli/generate.php]: Let me recomment to use these short aliases for bool flags only. It is not natural to execute CLI scripts like `php generate.php -r=example.yaml`

File: cli/generate.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    echo("\nRecipe not specified!\n");
+    echo $help;
+    die();
+}
+
+$recipefile = $options['recipe'];
+
+// Expanding '~' on Unix-like OS'es.
+if ($recipefile[0] === '~') {
+    $homedir = getenv('HOME');
```
Comment (mudrd8mz) [inline cli/generate.php]: I think I mentioned it before, we should check here that $homedir is not empty.

File: cli/generate.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+if ($recipefile === false) {
+    echo("\nInvalid recipe file!\n");
+    echo $help;
+    die();
+}
+
+$loglevels = Logger::getLevels();
+$loglevel = $options['loglevel'];
+if (!array_key_exists($loglevel, $loglevels)) {
+    echo("\nInvalid log level!\n");
```
Comment (mudrd8mz) [inline cli/generate.php]: We should display valid values, or at least the most common one (DEBUG, WARNING, INFO sounds like enough for this purpose)

**Comment by alexandru-elisei** [comment]
I'll create a patch to remedy the first two comments.

As for displaying valid log levels, I'll update the $help string to say what the valid log levels are.

File: cli/generate.php (hunk 2 above)
Comment (alexandru-elisei) [inline cli/generate.php]: There I'm simply expanding the path to the recipe file (and checking if the file exists/path is correct at line 87). I'm not doing anything yet about the target directory, as specifying one is not yet implemented.

Comment (mudrd8mz) [inline cli/generate.php]: I know. I am just thinking of OS or environments, where `getenv('HOME')` does not work. It seems to be cleaner to me to throw an error and inform the user. Anyway, it is not crucial.

Comment (alexandru-elisei) [inline cli/generate.php]: I see now, I've added the check to #32.

**Pull request merged**